perfil/core/management/commands/pre_calculate_stats.py

```diff
@@ -0,0 +1,48 @@
+import statistics
+from django.db import connection
+from django.core.management.base import BaseCommand
+
+from perfil.core.models import PreCalculatedStats
+
+
+class Command(BaseCommand):
+    help = "Pre calculate statistics over the politicians data to retrieve them faster"
+
+    model = PreCalculatedStats
+
+    @staticmethod
+    def _get_assets_median_per_year() -> dict:
+        sql = f"""
+            SELECT
+                core_candidate.year,
+                array_agg(core_asset.value) as assets_values
+            FROM core_asset
+            INNER JOIN core_candidate
+            ON core_candidate.id = core_asset.candidate_id
+            WHERE core_candidate.round_result LIKE 'ELEIT%'
+            GROUP BY core_candidate.year;
```

Comment:
Exato! Estamos pegando a mediana de todos os candidatos eleitos independentemente do cargo para pré-calcular a query. Ainda assim é possível separar por cargo no endpoint se quisermos, mas não é o que eu tinha planejado fazer no perfil-parlamentar hehehe lá eu estava pensando em pegar apenas as estatísticas gerais, do jeito que está. Eu lembrava de termos chega à conclusão de que poderia ser assim porque no fim não faria tanta diferença separar por cargos, mas fui procurar agora onde isso estava anotando e não achei em nenhum lugar. Acho que viajei 😅  

De toda forma, no perfil-parlamentar (atualmente) estamos deixando claro que as estatísticas mostradas são de todos candidatos eleitos no ano e não apenas dos candidatos de mesmo cargo :)

Mas posso mudar para separarmos por cargo! Vejo duas opções, qual acha que seria mais apropriada?
1. Usar o endpoint do jeito que está com o filtro no perfil-parlamentar sem mudar nada aqui (então calcularíamos na hora as estatísticas que queremos, sem usar as pré-calculadas - não demora tanto quanto o cálculo das estatísticas com todo mundo, pelo que eu me lembro, então acho que não seria um problema calcular na hora).
2. Usar o endpoint com o filtro no perfil-parlamentar e também mudar esse script para pré-calcularmos as estatísticas para os diferentes cargos (ficaria tudo meio parecido com o que temos hoje, poderíamos usar o mesmo modelo de `PreCalculatedStats` que temos e só mudar os valores de `type` para diferenciar a mediana de cada cargo).

---
### English
Exactly! We are taking the median of all elected candidates regardless of it's position to pre-calculate the query. It's still possible to filter by position in the endpoint if we want, but that's not what I was planning to do at perfil-parlamentar hehehe there I was just planning to use the general statistics, as it is. I remembered that we came to the conclusion that it could be like that because in the end it wouldn't make so much difference to separate by position, but now I went looking where I wrote that and I couldn't find anything. I think I just spaced out 😅 

Anyway, at perfil-parlamentar (currently) we are making clear that the stats shown are for all candidates elected in the year and not just candidates of the same position :)

But I can change to filter by position! I see two options, which do you think would be more appropriate?
1. Use the endpoint as it is with the filter in perfil-parlamentar without changing anything here (then we would calculate the stats we want on the fly, without using the pre-calculated ones - it doesn't take as long as calculating the stats for everyone, as I remember, so I guess it wouldn't be a problem to calculate it at runtime).
2. Use the endpoint with the filter on perfil-parlamentar but and also change this script here to pre-calculate the statistics for the different positions (the code would look similar to what we have now, we could use the same `PreCalculatedStats` model that we have and just change the `type` values to differentiate the median of each position).